### PR TITLE
Allow Complete Zipkin Url

### PR DESCRIPTION
**Pull request by lxghtless**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
Could you add some docs and tests for this?

**Comment by lxghtless** [comment]
> Could you add some docs and tests for this?

Absolutely.

File: index.js [hunk 1]
```diff
@@ -22,9 +22,17 @@ function zipkinPlugin (fastify, opts, next) {
 
   const ctxImpl = new CLSContext('zipkin')
 
+  let zipkinUrl = `${opts.zipkinUrl}/api/v2/spans`
```
Comment (mcollina) [inline index.js:25]: I would prefer to add a `zipkinPath` option instead so that things can be combined.

Comment (lxghtless) [reply index.js:25]: Sounds good. I'll make that change.

**Comment by jcchavezs** [comment]
I think we could just change it into httpReporterUrl like in
https://github.com/fastify/fastify-zipkin/issues/3

**Comment by lxghtless** [comment]
The way the module is currently set up makes it a bit awkward to test this feature... thinking through a way to not make the test feel hacky.

**Comment by lxghtless** [comment]
@mcollina -- what if we removed adding the path extension for the user?

> i.e.

**From**

```js
  const recorder = new BatchRecorder({
    logger: new HttpLogger({
      endpoint: `${opts.zipkinUrl}/api/v2/spans`,
      jsonEncoder: JSON_V2
    })
  })
```

**To**

```js
  const recorder = new BatchRecorder({
    logger: new HttpLogger({
      endpoint: opts.zipkinUrl,
      jsonEncoder: JSON_V2
    })
  })
```

It would be simpler and allow for use cases like NewRelic and Jaeger, but _would_ be a breaking change.

**Comment by mcollina** [comment]
That also works!

File: README.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ const fastify = require('fastify')()
 fastify.register(require('fastify-zipkin'), {
   serviceName: 'my-service-name',
   servicePort: 3000,
-  zipkinUrl: 'http://localhost:9411'
+  httpReporterUrl: 'http://localhost:9411/api/v2/spans'
```
Comment (jcchavezs) [inline README.md:21]: <3 thanks for this.

File: index.js [hunk 3]
```diff
@@ -18,21 +18,13 @@ const Instrumentation = zipkin.Instrumentation
...
-  // allow for user to provide complete url for
-  // use cases like using NewRelic Zipkin endpoints
-  if (opts.useCompleteZipkinUrl) {
-    zipkinUrl = opts.zipkinUrl
-  }
-
   const recorder = new BatchRecorder({
     logger: new HttpLogger({
-      endpoint: zipkinUrl,
+      endpoint: opts.httpReporterUrl,
```
Comment (mcollina) [inline index.js:27]: Let's call it `endpoint` and keep the same terminology of zipkin?

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by jcchavezs** [comment]
Zipkin terminology is httpReporterUrl, endpoint is a span concept.

File: index.js (hunk 3 above)
Comment (lxghtless) [reply index.js:27]: I could be convinced either way, but I _think_ @jcchavezs is correct in that Zipkin calls this `httpReporterUrl`

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by lxghtless** [comment]
Does anyone know when this might get merged?

**Comment by lxghtless** [comment]
> Does anyone know when this might get merged?

I'd be happy to help in any way if needed. I appreciate the library and the effort to maintain/publish it.

**Pull request merged**